docs/user_guide/examples.md

```diff
@@ -1,14 +1,248 @@
+
+This is implemented by (additionally) inheriting from {external+law:py:class}`~law.tasks.RunOnceTask`.
+Moreover, to mark the task as complete, the {external=law:py:meth}`~law.tasks.RunOnceTask.mark_complete()` method is to be added at the end of the ```run()``` method after the logic of the task is executed, or the complete```run()``` method is either to be wrapped by the {external=law:py:meth}`~law.tasks.RunOnceTask.complete_on_success` decorator.
+One must be careful with the latter, since the decorator will mark the task as complete even if it fails.
+Therefore, it is recommended to also add a check at the end and raise an error in case the task fails.
+
+```python
+import law
+
+from columnflow.tasks.framework.base import AnalysisTask
+
+class MyPrintingTask(AnalysisTask, law.tasks.RunOnceTask):
+
+    @law.tasks.RunOnceTask.complete_on_success
+    def run(self):
+        # Do something here
+        if fail:
+            raise Exception("this task just failed")
+
+    # Alternative
+    def run(self):
+        # Do something here
+        self.mark_complete()
+```
+
+### Add the task to the analysis
+
+After creating a new task, make sure to add the new task to the law config file in your analysis folder, in order to run the task.
+The new task has to be specified under the ```[outputs]``` section as follows:
+
+```ini
+[outputs]
+
+lfn_sources: wlcg_fs_desy_store
+
+cf.MyNewTask: wlcg
+...
+```
+### Executing the Task
+
+after sourcing the law environment, the task can be executed using the command:
+```bash
+law run cf.MyTask --param1 foo
+```
+The parameters can be given using the syntax ```--param1 foo```, where ```param1``` is the name of the parameter and ```foo``` is the value of the parameter.
+If the parameter's name include an underscore, the underscore is replaced by a hyphen (```--param_1 foo``` -> ```--param-1 foo```).
+If the parameter is a boolean, the parameter can be set to ```True``` by only typing ```--param-boolean```.
+If the parameter is a list, the values can be given as a comma-separated list (```--param2 foo,bar,baz```).
 
 ## Custom law.cfg file for users without grid certificate (CMS specific?)
 
-TODO
+Users, who do not have a grid certificate, are not able to run jobs on the grid or store and fetch data from it.
+However, they can still run jobs locally.
+For this purpose, a custom law config file should be created in the analysis folder, which specifies a local dcache and the path to the nanoAoDs for every required campaing, and set the output locations for the tasks to local.
+It is recommended to create a copy of the default law config file and modify it as needed and then redirect the setup to the custom config file by setting the ```LAW_CONFIG_FILE``` environment variable in the `setup.sh` file to the path of the custom config file as follows:
+
+```bash
+export LAW_CONFIG_FILE="${LAW_CONFIG_FILE:-${ANALYSIS_DIR}/law.nocert.cfg}"
+```
+
+The costum config file should contain the following sections:
```

Comment:
add link to config documentation? https://law.readthedocs.io/en/latest/config.html